derive_client/clients/base_client.py

```diff
@@ -155,8 +155,17 @@ def deposit_to_derive(self, chain_id: ChainID, currency: Currency, amount: float
                 receiver=receiver,
                 chain_id=chain_id,
             )
-
+        # Hmmmm this is a bit of a hack, but we need to derive the token data
```

Comment:
THis is now unecessary, was trying to isolate exactly where to hook in. 

Where would be better to do this?